File: gluecodium/test/dart/lib/src/test/test_class_enum_value_with_under_score.dart


import 'dart:ffi';
import 'package:library/src/_library_context.dart' as __lib;
import 'package:library/src/_native_base.dart' as __lib;
import 'package:library/src/_token_cache.dart' as __lib;

/// Test class for cross references of enum values with under score.
///
/// Some cases don't work.
abstract class TestClassEnumValueWithUnderScore {

  /// Note:a [Error.invalidParameter] error is generated.
  ///
  void fooGoodCase();
  /// Note:* [Error.INVALID_PARAMETER] error is generated.
  ///
  void fooStarBadCase1();
  /// *Note: [Error.INVALID_PARAMETER] error is generated.
  ///
  void fooStarBadCase2();
  /// This is a test
  ///
  /// *Note: [Error.invalidParameter] is generated.
  ///
  void fooSentenceWithANewLineGoodCase();
  /// This is a tes
  ///
  /// *Note: [Error.INVALID_PARAMETER] is generated.
  ///
  void fooShorterSentenceWithANewLineBadCase();
}


// TestClassEnumValueWithUnderScore "private" section, not exported.

final _testTestclassenumvaluewithunderscoreRegisterFinalizer = __lib.catchArgumentError(() => __lib.nativeLibrary.lookupFunction<
    Void Function(Pointer<Void>, Int32, Handle),
    void Function(Pointer<Void>, int, Object)
  >('library_test_TestClassEnumValueWithUnderScore_register_finalizer'));
final _testTestclassenumvaluewithunderscoreCopyHandle = __lib.catchArgumentError(() => __lib.nativeLibrary.lookupFunction<
    Pointer<Void> Function(Pointer<Void>),
    Pointer<Void> Function(Pointer<Void>)
  >('library_test_TestClassEnumValueWithUnderScore_copy_handle'));
final _testTestclassenumvaluewithunderscoreReleaseHandle = __lib.catchArgumentError(() => __lib.nativeLibrary.lookupFunction<
    Void Function(Pointer<Void>),
    void Function(Pointer<Void>)
  >('library_test_TestClassEnumValueWithUnderScore_release_handle'));







class TestClassEnumValueWithUnderScore$Impl extends __lib.NativeBase implements TestClassEnumValueWithUnderScore {

  TestClassEnumValueWithUnderScore$Impl(Pointer<Void> handle) : super(handle);

  @override
  void fooGoodCase() {
    final _fooGoodCaseFfi = __lib.catchArgumentError(() => __lib.nativeLibrary.lookupFunction<Void Function(Pointer<Void>, Int32), void Function(Pointer<Void>, int)>('library_test_TestClassEnumValueWithUnderScore_fooGoodCase'));
    final _handle = this.handle;
    _fooGoodCaseFfi(_handle, __lib.LibraryContext.isolateId);

  }

  @override
  void fooStarBadCase1() {
    final _fooStarBadCase1Ffi = __lib.catchArgumentError(() => __lib.nativeLibrary.lookupFunction<Void Function(Pointer<Void>, Int32), void Function(Pointer<Void>, int)>('library_test_TestClassEnumValueWithUnderScore_fooStarBadCase1'));
    final _handle = this.handle;
    _fooStarBadCase1Ffi(_handle, __lib.LibraryContext.isolateId);

  }

  @override
  void fooStarBadCase2() {
    final _fooStarBadCase2Ffi = __lib.catchArgumentError(() => __lib.nativeLibrary.lookupFunction<Void Function(Pointer<Void>, Int32), void Function(Pointer<Void>, int)>('library_test_TestClassEnumValueWithUnderScore_fooStarBadCase2'));
    final _handle = this.handle;
    _fooStarBadCase2Ffi(_handle, __lib.LibraryContext.isolateId);

  }

  @override
  void fooSentenceWithANewLineGoodCase() {
    final _fooSentenceWithANewLineGoodCaseFfi = __lib.catchArgumentError(() => __lib.nativeLibrary.lookupFunction<Void Function(Pointer<Void>, Int32), void Function(Pointer<Void>, int)>('library_test_TestClassEnumValueWithUnderScore_fooSentenceWithANewLineGoodCase'));
    final _handle = this.handle;
    _fooSentenceWithANewLineGoodCaseFfi(_handle, __lib.LibraryContext.isolateId);

  }

  @override
  void fooShorterSentenceWithANewLineBadCase() {
    final _fooShorterSentenceWithANewLineBadCaseFfi = __lib.catchArgumentError(() => __lib.nativeLibrary.lookupFunction<Void Function(Pointer<Void>, Int32), void Function(Pointer<Void>, int)>('library_test_TestClassEnumValueWithUnderScore_fooShorterSentenceWithANewLineBadCase'));
    final _handle = this.handle;
    _fooShorterSentenceWithANewLineBadCaseFfi(_handle, __lib.LibraryContext.isolateId);

  }


}

Pointer<Void> testTestclassenumvaluewithunderscoreToFfi(TestClassEnumValueWithUnderScore value) =>
  _testTestclassenumvaluewithunderscoreCopyHandle((value as __lib.NativeBase).handle);

TestClassEnumValueWithUnderScore testTestclassenumvaluewithunderscoreFromFfi(Pointer<Void> handle) {
  if (handle.address == 0) throw StateError("Expected non-null value.");
  final instance = __lib.getCachedInstance(handle);
  if (instance != null && instance is TestClassEnumValueWithUnderScore) return instance;

  final _copiedHandle = _testTestclassenumvaluewithunderscoreCopyHandle(handle);
  final result = TestClassEnumValueWithUnderScore$Impl(_copiedHandle);
  __lib.cacheInstance(_copiedHandle, result);
  _testTestclassenumvaluewithunderscoreRegisterFinalizer(_copiedHandle, __lib.LibraryContext.isolateId, result);
  return result;
}

void testTestclassenumvaluewithunderscoreReleaseFfiHandle(Pointer<Void> handle) =>
  _testTestclassenumvaluewithunderscoreReleaseHandle(handle);

Pointer<Void> testTestclassenumvaluewithunderscoreToFfiNullable(TestClassEnumValueWithUnderScore? value) =>
  value != null ? testTestclassenumvaluewithunderscoreToFfi(value) : Pointer<Void>.fromAddress(0);

TestClassEnumValueWithUnderScore? testTestclassenumvaluewithunderscoreFromFfiNullable(Pointer<Void> handle) =>
  handle.address != 0 ? testTestclassenumvaluewithunderscoreFromFfi(handle) : null;

void testTestclassenumvaluewithunderscoreReleaseFfiHandleNullable(Pointer<Void> handle) =>
  _testTestclassenumvaluewithunderscoreReleaseHandle(handle);

// End of TestClassEnumValueWithUnderScore "private" section.

File: gluecodium/test/swift/test/TestClassEnumValueWithUnderScore.swift
//

//

import Foundation

/// Test class for cross references of enum values with under score. Some cases don't work.
public class TestClassEnumValueWithUnderScore {


    let c_instance : _baseRef

    init(cTestClassEnumValueWithUnderScore: _baseRef) {
        guard cTestClassEnumValueWithUnderScore != 0 else {
            fatalError("Nullptr value is not supported for initializers")
        }
        c_instance = cTestClassEnumValueWithUnderScore
    }

    deinit {
        test_TestClassEnumValueWithUnderScore_remove_swift_object_from_wrapper_cache(c_instance)
        test_TestClassEnumValueWithUnderScore_release_handle(c_instance)
    }

    /// Note:a `Error.invalidParameter` error is generated.
    public func fooGoodCase() -> Void {
        test_TestClassEnumValueWithUnderScore_fooGoodCase(self.c_instance)
    }
    /// Note:* [Error.INVALID_PARAMETER] error is generated.
    public func fooStarBadCase1() -> Void {
        test_TestClassEnumValueWithUnderScore_fooStarBadCase1(self.c_instance)
    }
    /// *Note: [Error.INVALID_PARAMETER] error is generated.
    public func fooStarBadCase2() -> Void {
        test_TestClassEnumValueWithUnderScore_fooStarBadCase2(self.c_instance)
    }
    /// This is a test
    ///
    /// *Note: `Error.invalidParameter` is generated.
    public func fooSentenceWithANewLineGoodCase() -> Void {
        test_TestClassEnumValueWithUnderScore_fooSentenceWithANewLineGoodCase(self.c_instance)
    }
    /// This is a tes
    ///
    /// *Note: [Error.INVALID_PARAMETER] is generated.
    public func fooShorterSentenceWithANewLineBadCase() -> Void {
        test_TestClassEnumValueWithUnderScore_fooShorterSentenceWithANewLineBadCase(self.c_instance)
    }

}



internal func getRef(_ ref: TestClassEnumValueWithUnderScore?, owning: Bool = true) -> RefHolder {
    guard let c_handle = ref?.c_instance else {
        return RefHolder(0)
    }
    let handle_copy = test_TestClassEnumValueWithUnderScore_copy_handle(c_handle)
    return owning
        ? RefHolder(ref: handle_copy, release: test_TestClassEnumValueWithUnderScore_release_handle)
        : RefHolder(handle_copy)
}

extension TestClassEnumValueWithUnderScore: NativeBase {
    /// :nodoc:
    var c_handle: _baseRef { return c_instance }
}
extension TestClassEnumValueWithUnderScore: Hashable {
    /// :nodoc:
    public static func == (lhs: TestClassEnumValueWithUnderScore, rhs: TestClassEnumValueWithUnderScore) -> Bool {
        return lhs.c_handle == rhs.c_handle
    }

    /// :nodoc:
    public func hash(into hasher: inout Hasher) {
        hasher.combine(c_handle)
    }
}

internal func TestClassEnumValueWithUnderScore_copyFromCType(_ handle: _baseRef) -> TestClassEnumValueWithUnderScore {
    if let swift_pointer = test_TestClassEnumValueWithUnderScore_get_swift_object_from_wrapper_cache(handle),
        let re_constructed = Unmanaged<AnyObject>.fromOpaque(swift_pointer).takeUnretainedValue() as? TestClassEnumValueWithUnderScore {
        return re_constructed
    }
    let result = TestClassEnumValueWithUnderScore(cTestClassEnumValueWithUnderScore: test_TestClassEnumValueWithUnderScore_copy_handle(handle))
    test_TestClassEnumValueWithUnderScore_cache_swift_object_wrapper(handle, Unmanaged<AnyObject>.passUnretained(result).toOpaque())
    return result
}

internal func TestClassEnumValueWithUnderScore_moveFromCType(_ handle: _baseRef) -> TestClassEnumValueWithUnderScore {
    if let swift_pointer = test_TestClassEnumValueWithUnderScore_get_swift_object_from_wrapper_cache(handle),
        let re_constructed = Unmanaged<AnyObject>.fromOpaque(swift_pointer).takeUnretainedValue() as? TestClassEnumValueWithUnderScore {
        test_TestClassEnumValueWithUnderScore_release_handle(handle)
        return re_constructed
    }
    let result = TestClassEnumValueWithUnderScore(cTestClassEnumValueWithUnderScore: handle)
    test_TestClassEnumValueWithUnderScore_cache_swift_object_wrapper(handle, Unmanaged<AnyObject>.passUnretained(result).toOpaque())
    return result
}

internal func TestClassEnumValueWithUnderScore_copyFromCType(_ handle: _baseRef) -> TestClassEnumValueWithUnderScore? {
    guard handle != 0 else {
        return nil
    }
    return TestClassEnumValueWithUnderScore_moveFromCType(handle) as TestClassEnumValueWithUnderScore
}
internal func TestClassEnumValueWithUnderScore_moveFromCType(_ handle: _baseRef) -> TestClassEnumValueWithUnderScore? {
    guard handle != 0 else {
        return nil
    }
    return TestClassEnumValueWithUnderScore_moveFromCType(handle) as TestClassEnumValueWithUnderScore
}

internal func copyToCType(_ swiftClass: TestClassEnumValueWithUnderScore) -> RefHolder {
    return getRef(swiftClass, owning: false)
}

internal func moveToCType(_ swiftClass: TestClassEnumValueWithUnderScore) -> RefHolder {
    return getRef(swiftClass, owning: true)
}

internal func copyToCType(_ swiftClass: TestClassEnumValueWithUnderScore?) -> RefHolder {
    return getRef(swiftClass, owning: false)
}

internal func moveToCType(_ swiftClass: TestClassEnumValueWithUnderScore?) -> RefHolder {
    return getRef(swiftClass, owning: true)
}



